src/init.cpp

```diff
@@ -350,6 +351,7 @@ void DynamicDataLoader::initialize()
     add( "city_building", &city_buildings::load );
     add( "map_extra", &MapExtras::load );
 
+    add( "weather_type", &weather::load_weather_type );
```

Comment:
You also need to add a function to reset (unload) the weather types in ` DynamicDataLoader::unload_data`.